src/GraphQL.Tests/DI/DefaultServiceProviderTests.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.DI
+{
+    public class DefaultServiceProviderTests
+    {
+        private readonly IServiceProvider _serviceProvider = new DefaultServiceProvider();
+
+        [Fact]
+        public void throws_for_null_service_type()
+        {
+            Should.Throw<ArgumentNullException>(() => _serviceProvider.GetService(null));
+        }
+
+        [Fact]
+        public void returns_self_for_iserviceprovider()
+        {
+            _serviceProvider.GetService(typeof(IServiceProvider)).ShouldBe(_serviceProvider);
+        }
+
+        [Fact]
+        public void returns_null_for_interfaces()
```

Comment:
???

Reply:
It is impossible to create an instance of an interface. So it threw an exception, which breaks Schema, when it tries to request `IEnumerable<…>`. 

GetService should return null if it cannot create the instance. (At least that is how it is usually works.) So I had to change the DefaultServiceProvider to return null for interface requests. Recently had to make a similar change in the conventions project for the same reason.

Reply:
I was wrong.